test/aievec/conv2d_uij_i8_noinit_aie-ml.mlir

```diff
@@ -0,0 +1,94 @@
+
+            //Third row
+            //first point
+            %a31 = affine.load %A[%arg3+2, %arg4+0] : memref<18x288xi8>
+            %b31 = affine.load %B[32] : memref<48xi8>
+            %p31 = arith.muli %a31, %b31 : i8
+            %c31 = arith.addi %c23, %p31 : i8
+
+            //second point
+            %a32 = affine.load %A[%arg3+2, %arg4+1] : memref<18x288xi8>
+            %b32 = affine.load %B[34] : memref<48xi8>
+            %p32 = arith.muli %a32, %b32 : i8
+            %c32 = arith.addi %c31, %p32 : i8
+
+            //third point
+            %a33 = affine.load %A[%arg3+2, %arg4+2] : memref<18x288xi8>
+            %b33 = affine.load %B[36] : memref<48xi8>
+            %p33 = arith.muli %a33, %b33 : i8
+            %c33 = arith.addi %c32, %p33 : i8
+
+            //Store accumulated sum
+            affine.store %c33, %C[%arg3, %arg4] : memref<16x256xi8>
+        }
+    }
+    return
+}
+
+//CHECK-NEXT:    %c0 = arith.constant 0 : index
+//CHECK-NEXT:    %0 = aievec.upd %arg1[%c0] {index = 0 : i8, offset = 0 : si32} : memref<48xi8>, vector<64xi8>
+//CHECK-NEXT:    %1 = aievec.shuffle %0 {mode = 0 : ui32} : vector<64xi8>, vector<64xi8>
+//CHECK-NEXT:    %2 = aievec.shift %1 {shift = 8 : si32} : vector<64xi8>, vector<64xi8>
+//CHECK-NEXT:    %3 = aievec.shift %1 {shift = 16 : si32} : vector<64xi8>, vector<64xi8>
+//CHECK-NEXT:    %c0_0 = arith.constant 0 : index
+//CHECK-NEXT:    %c16 = arith.constant 16 : index
+//CHECK-NEXT:    %c1 = arith.constant 1 : index
+//CHECK-NEXT:    scf.for %arg3 = %c0_0 to %c16 step %c1 {
+//CHECK-NEXT:      %c1_1 = arith.constant 1 : index
+//CHECK-NEXT:      %4 = arith.addi %arg3, %c1_1 : index
+//CHECK-NEXT:      %c2 = arith.constant 2 : index
+//CHECK-NEXT:      %5 = arith.addi %arg3, %c2 : index
+//CHECK-NEXT:      %c0_2 = arith.constant 0 : index
+//CHECK-NEXT:      %c256 = arith.constant 256 : index
+//CHECK-NEXT:      %c32 = arith.constant 32 : index
+//CHECK-NEXT:      scf.for %arg4 = %c0_2 to %c256 step %c32 {
+//CHECK-NEXT:        %6 = aievec.upd %arg0[%arg3, %arg4] {index = 0 : i8, offset = 0 : si32} : memref<18x288xi8>, vector<64xi8>
+//CHECK-NEXT:        %7 = aievec.mul_conv %6, %1 {M = 32 : si32, N = 8 : si32} : vector<64xi8>, vector<64xi8>, vector<32xi32>
+//CHECK-NEXT:        %8 = aievec.upd %arg0[%4, %arg4] {index = 0 : i8, offset = 0 : si32} : memref<18x288xi8>, vector<64xi8>
+//CHECK-NEXT:        %9 = aievec.fma_conv %8, %2, %7 {M = 32 : si32, N = 8 : si32} : vector<64xi8>, vector<64xi8>, vector<32xi32>
+//CHECK-NEXT:        %10 = aievec.upd %arg0[%5, %arg4] {index = 0 : i8, offset = 0 : si32} : memref<18x288xi8>, vector<64xi8>
+//CHECK-NEXT:        %11 = aievec.fma_conv %10, %3, %9 {M = 32 : si32, N = 8 : si32} : vector<64xi8>, vector<64xi8>, vector<32xi32>
+//CHECK-NEXT:        %12 = aievec.srs %11 {shift = 0 : i8} : vector<32xi32>, vector<32xi8>
+//CHECK-NEXT:        vector.transfer_write %12, %arg2[%arg3, %arg4] {in_bounds = [true]} : vector<32xi8>, memref<16x256xi8>
```

Comment:
To future-proof the test, use FileCheck regex matching/substitution blocks as opposed to a code dump. This test fixes the output, rather than ensuring semantic correctness.

Reply:
You can use regex matching and substitutions everywhere, and still check the details of each operation. You just avoid relying on naming specifics, and ideally the order of operations, focusing on dependencies and semantics only. Check, for instance, this MLIR test: [vector-contract-matvec-transforms.mlir](https://github.com/llvm/llvm-project/blob/main/mlir/test/Dialect/Vector/vector-contract-matvec-transforms.mlir).

We should do this in all our tests, we haven't. At some point we need to bring that up to standard but, for now, I think a good step in the right direction is making all new tests to align with MLIR standards.